### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by eriknewland**

### CODE REVIEW

File: client/src/components/ratings/Ratings.jsx [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+import axios from 'axios';
 
 export default function Ratings({
   feature
 }) {
 
+  const [reviews, setReviews] = useState([]);
+
+  useEffect(() => {
+    fetcher.ratings.getReviews(40348)
```
Comment (boltonlin) [inline client/src/components/ratings/Ratings.jsx:14]: should get reviews for featured product via `feature.id`

File: client/src/components/ratings/Ratings.jsx [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+      .then(({data}) => setReviews([...data.results]))
+      .catch((error) => console.log(error));
+  }, []);
```
Comment (boltonlin) [inline client/src/components/ratings/Ratings.jsx:17]: this use effect should be on change of feature

File: client/src/components/ratings/sub-components/ReviewTile.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    let date = new Date(dateString);
+    let mm = date.getMonth() + 1;
+    let dd = date.getDate();
+    let yyyy = date.getFullYear();
+    return `${mm}/${dd}/${yyyy}`;
+  };
+
+  return (
+    <div>
+      {console.log(review)}
```
Comment (boltonlin) [inline client/src/components/ratings/sub-components/ReviewTile.jsx:23]: do you still need this? should delete later if so. maybe add todo if you need this?

File: server/controllers/reviews.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
 const model = require('../models').reviews;
 
 module.exports = {
-
+  getReviews: (req, res) => {
+    model.getReviews(req.params.product_id)
+      .then(results => {
+        console.log(results);
```
Comment (boltonlin) [inline server/controllers/reviews.js]: don't need this
```suggestion
        .then(results => res.status(200).send(results.data))
```

File: server/models/reviews.js [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
 const authHeader = {
   'Authorization': process.env.TOKEN
 };
 
 module.exports = {
-
+  getReviews: (id) => {
+    //console.log(id);
+    let options = {
+      url: endpoint,
```
Comment (boltonlin) [inline server/models/reviews.js]:
```suggestion
      url: endpoint + '/'
```

File: server/models/reviews.js [hunk 6]
```diff
@@ -1,10 +1,22 @@
 const axios = require('axios');
 
-const endpoint = process.env.BASEURL + '/reviews';
+const endpoint = process.env.BASEURL + '/reviews/';
```
Comment (boltonlin) [inline server/models/reviews.js]:
```suggestion
const endpoint = process.env.BASEURL + '/reviews';
```
reason: not all of these api requests will be `/reviews/`, but all these api requests will start with `/reviews`

**Review by boltonlin — CHANGES_REQUESTED** [review]
looks good. not sure if all the logs are necessary so made suggestions to clean.

File: client/src/components/ratings/Ratings.jsx (hunk 1 above)
Comment (eriknewland) [reply client/src/components/ratings/Ratings.jsx:14]: Got it - used this as a placeholder until we get id from top level of app

File: client/src/components/ratings/sub-components/ReviewTile.jsx (hunk 3 above)
Comment (eriknewland) [reply client/src/components/ratings/sub-components/ReviewTile.jsx:23]: Bad habit of mine to leave in old logs. Will clean up

File: server/models/reviews.js (hunk 6 above)
Comment (eriknewland) [reply server/models/reviews.js]: Understood, thank you for the clarification!

**Comment by eriknewland** [comment]
merge difficulties, trying again later

**Pull request closed without merging**